blivet/fstab.py

```diff
@@ -0,0 +1,790 @@
+
+        affected_params = [self._entry.source,
+                           self._entry.target,
+                           self._entry.fstype,
+                           self._entry.options,
+                           self._entry.comment]
+
+        for param in affected_params:
+            if param is None:
+                param = ""
+
+    @property
+    def entry(self):
+        return self._entry
+
+    @entry.setter
+    def entry(self, value):
+        """ Setter for the whole internal entry value
+
+            :param value: fstab entry
+            :type value: :class: `libmount.Fs`
+        """
+        self._entry = value
+
+    @property
+    def spec(self):
+        return self._entry.source if self._entry.source != "" else None
+
+    @spec.setter
+    def spec(self, value):
+        self._entry.source = value if value is not None else ""
+
+    @property
+    def file(self):
+        return self._entry.target if self._entry.target != "" else None
+
+    @file.setter
+    def file(self, value):
+        self._entry.target = value if value is not None else ""
+
+    @property
+    def vfstype(self):
+        return self._entry.fstype if self._entry.fstype != "" else None
+
+    @vfstype.setter
+    def vfstype(self, value):
+        self._entry.fstype = value if value is not None else ""
+
+    @property
+    def mntops(self):
+        """ Return mount options
+
+            :returns: list of mount options or None when not set
+            :rtype: list of str
+        """
+
+        if self._entry.options == "":
+            return None
+
+        return self._entry.options.split(',')
+
+    def get_raw_mntops(self):
+        """ Return mount options
+
+            :returns: comma separated string of mount options or None when not set
+            :rtype: str
+        """
+
+        return self._entry.options if self._entry.options != "" else None
+
+    @mntops.setter
+    def mntops(self, values):
+        """ Set new mount options from the list of strings
+
+            :param values: mount options (see man fstab(5) fs_mntops)
+            :type values: list of str
+        """
+
+        # libmount.Fs() internally stores options as a comma separated string
+        if values is None:
+            self._entry.options = ""
+        else:
+            self._entry.options = ','.join([x for x in values if x != ""])
+
+    def mntops_add(self, values):
+        """ Append new mount options to already existing ones
+
+            :param values: mount options (see man fstab(5) fs_mntops)
+            :type values: list of str
+        """
+
+        self._entry.append_options(','.join([x for x in values if x != ""]))
+
+    @property
+    def freq(self):
+        return self._entry.freq
+
+    @freq.setter
+    def freq(self, value):
+        self._entry.freq = value
+
+    @property
+    def passno(self):
+        return self._entry.passno
+
+    @passno.setter
+    def passno(self, value):
+        self._entry.passno = value
```

Comment:
My recollection is that the only valid values for `passno` are 0, 1, 2. Should this be checked/enforced here. Perhaps somewhere else?

Reply:
Fixed.